providers/oidc.go

```diff
@@ -281,7 +297,19 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 type OIDCClaims struct {
 	rawClaims         map[string]interface{}
 	UserID            string
-	Subject           string `json:"sub"`
-	Verified          *bool  `json:"email_verified"`
-	PreferredUsername string `json:"preferred_username"`
+	Groups            []string `json:"groups"`
+	Subject           string   `json:"sub"`
+	Verified          *bool    `json:"email_verified"`
+	PreferredUsername string   `json:"preferred_username"`
+}
+
+func (c *OIDCClaims) containsGroups(groups []string) bool {
```

Comment:
I think this is perfectly fine for our scenarios where the array sizes are small.

I think about ~100 is usually the threshold where O(N^2) double loop array operations are still essentially constant time.

If you can think of scenarios where we might be eclipsing that, we should potentially use sets & set intersection to get this down to O(N).

Reply:
@NickMeves changed this to populate a map then use that, better?